src/main/resources/db/migration/V17_1__STAC_Item_Search_-_Create_STAC_Collections_partitioned_table.sql

```diff
@@ -0,0 +1,22 @@
+CREATE TABLE stac_collections_part (
+    id varchar(100),
+    bbox double precision[],
+    properties jsonb,
+    collection_id uuid REFERENCES collections_details (id),
+    geom geometry(Geometry, 4326),
+    p_id int GENERATED ALWAYS AS IDENTITY
+)
+PARTITION BY LIST (collection_id);
+
+-- Add primary key on p_id and collection_id
+
+ALTER TABLE stac_collections_part ADD PRIMARY KEY (p_id, collection_id);
+
+-- Add indexes on id, geom and properties
+
+CREATE INDEX idx_stac_collections_part_id ON stac_collections_part (id);
+CREATE INDEX idx_stac_collections_part_geom ON stac_collections_part USING GIST (geom);
+CREATE INDEX idx_stac_collections_part_properties ON stac_collections_part USING GIN (properties);
+
```

Comment:
Should we also add a `bbox` index as well? I believe the `bbox` is also used as a param in the query for ItemSearch? We can add a _GIN_ index for the `bbox` column

Reply:
`bbox` is not used for querying, only `geom` is used

Reply:
https://github.com/radiantearth/stac-api-spec/tree/release/v1.0.0/item-search#query-parameter-table

Reply:
You're right, `bbox` is a query parameter to find all items contained in the bbox. But currently, the `bbox` column is not used to determine that. The `geom` column is used.

If `bbox` is included in the query params, the `FeatureQueryBuilder.setBbox()` method is used, which uses `geom` - https://github.com/datakaveri/ogc-resource-server/pull/221/files#diff-86d87d89fcf0ed1bc87b79ceeb08f231f2b7efb721744f5c05f3d6ca0b363e82R495 , https://github.com/datakaveri/ogc-resource-server/pull/221/files#diff-0528901be6c607b0ab42b5fcdc9777099d12c78392cb602a2ff1b14e0b0f6e48R77

If there is a way to use `bbox` for this query that is better/more performant, we can try that. using PostGIS functions or some array comparisons

Reply:
I was thinking using the bbox itself as a parameter since we have those values in the table, but you're right, we need to test it out extensively to even consider it as an option. I'll close this conversation. Will approve the changes.